### ISSUE TITLE
Can we disable the auto-centering that occurs onRegionChangeComplete?

### ISSUE THREAD

**Issue by RyanTG** [react-native-maps/react-native-maps#3692]
# Question

When `onRegionChangeComplete` fires, it seems to auto-center the map. If I stop panning/dragging, and then I quickly pan/drag again while that function is firing, once it is done then map extent jumps back to where it was when I first stopped panning/dragging. This is problematic, and we are having trouble getting around it. **Is it possible to disable this behavior?**

I notice on pro map apps like Zillow, the new data is silently pulled when you move the map (when you let go of pan/drag), without an auto-center. On the other hand, I was surprised to see that Airbnb doesn't auto-refresh the map when you move it! Every time you move the map, a button pops up that says, "search this area". So you have to click that each time you want results. Given that react-native-maps is licensed by Airbnb, perhaps auto-refreshing data on map move isn't what this package excels at? **Edit:  was just reminded that airbnb no longer uses react native! ;p**

This issue has come up a bunch on here. Usually the response is about not keeping region in state.

Is there anything else we're missing? We've been stumped on this for a while! Thank you!

Our mapview (we have it not fire unless you move the map more than a certain distance):

````
<MapView
                        ref={this.mapRef}
                        region={{
                            latitude,
                            longitude,
                            latitudeDelta,
                            longitudeDelta,
                        }}
                        style={s.map}
                        onRegionChangeComplete={this.onRegionChange}
                        showsUserLocation={true}
                        moveOnMarkerPress={false}
                        showsMyLocationButton={false}
                        provider = { MapView.PROVIDER_GOOGLE }
                        customMapStyle={theme.theme === 'dark' ? androidCustomDark : []}
                    >
                        {mapLocations.map(l => <CustomMarker key={l.id} marker={l} navigation={navigation} s={s} />)}
</MapView>

    onRegionChange = (region) => {
        const compareRegion = (region) => {
            this.props.getLocationsConsideringZoom(region.latitude, region.longitude, region.latitudeDelta, region.longitudeDelta)
        }

        if (Math.abs(region.latitude - this.prevRegion.latitude) > 0.001) {
            setTimeout(compareRegion, 800, region)
            this.props.updateMapCoordinates({
                ...region
            })
        }
        this.prevRegion = region
    }
````
And our project: https://github.com/bpoore/pbm-react
Happy to give any more details.

**Comment by RyanTG** [comment]
Closing this since it didn't get any attention. This issue has come up many times here, and seems sorta unresolved (or the resolution is vague).

On our project, we're going to disable onRegionChange, and instead add a "search this area" button whenever the map is moved, so the user can manually reload the data for that extent.

**Comment by tgreco** [comment]
I don't think this issue should have been closed.
I would love a resolution.

**Comment by RyanTG** [comment]
You're right. I will re-open. This still seems valid.

**Comment by tabish-kook** [comment]
Anyone found the solution of this problem?

**Comment by hsource** [comment]
For many people, https://github.com/react-native-maps/react-native-maps/issues/199#issuecomment-330901293 should provide an option.

TL;DR: set `moveOnMarkerPress={false}`
